File: src/main/java/kafkastreams/javaexercises/Exercise_0_HelloKafkaStreams.java
package kafkastreams.javaexercises;

import org.apache.kafka.common.serialization.Serde;
import org.apache.kafka.common.serialization.Serdes;
import org.apache.kafka.streams.StreamsBuilder;
import org.apache.kafka.streams.kstream.*;

public class Exercise_0_HelloKafkaStreams {

    private static Serde<String> strings = Serdes.String();

    /**
     * Read the Kafka topic 'text' and send the contents directly to
     * the new topic 'pass-through'
     */
    public void passEventsThroughDirectly(StreamsBuilder builder) {

    }

}

File: src/main/scala/kafkastreams/scalaexercises/Exercise_0_HelloKafkaStreams.scala
package kafkastreams.scalaexercises

import org.apache.kafka.streams.scala.ImplicitConversions._
import org.apache.kafka.streams.scala.Serdes.String
import org.apache.kafka.streams.scala.StreamsBuilder

class Exercise_0_HelloKafkaStreams {

  /**
    * Read the Kafka topic 'text' and send the contents directly to
    * the new topic 'pass-through'
    */
  def passEventsThroughDirectly(builder: StreamsBuilder): Unit = {
    val stream = builder.stream[String, String]("text")
    stream.to("pass-through")
  }

}
